### PR TITLE
FIX: recipients index

### LINKED ISSUE #2362: Batch Send UI bugs
* [ ]  when adding a recipient, scroll scrolls to _second to last_ recipient, not the last one (off-by-one bug?)
* [ ] when scanning address for a recipient scroll resets to the first recipient
* [ ] when removing _last_ recipient, scroll renders just blank space, you need to manually scroll to your other recipients

### dire need for e2e test of this feature!

since we have a huge overhead on running an e2e testcase and importing a wallet, I suggest e2e test for batch send will be added to the end of `can import BIP84 mnemonic, fetch balance & transactions, then create a transaction` test. do several `device.pressBack()` so app goes back to main screen, and start automating e2e from there. to speed things up you can temporarily comment out parts of this test (basically start coding test right after `helperImportWallet()`)

### PR DESCRIPTION
**Pull request by limpbrains**
- Fixes white screen when remove last of 3 recipients on android
- Add key for addresses array so that FlatList shows correct animations
- Fixes track of current recipient index in Component store. Recepient remove now correctly removes active element, not the last one.

closes #2362

### CODE REVIEW

**Review by marcosrdz — APPROVED** [review]
Ahhh...the beauty of useState not updating. Classic

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/2w88duhxzp72kequ7hpw0u22kc

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-sendmany-index-1614869952.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ehc94fec4euqn77j8j6xmhkbg8

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-sendmany-index-1614885653.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/5w8wky1mypbbvwvm6hpfv9pmk0

[download apk](http://filestorage.bluewallet.io:1488/limpbrains-sendmany-index-1614928966.apk)

**Comment by limpbrains** [comment]
@Overtorment merge it, it's ready

File: tests/e2e/bluewallet.spec.js [hunk 1]
```diff
@@ -282,7 +282,7 @@ describe('BlueWallet UI Tests', () => {
     process.env.TRAVIS && require('fs').writeFileSync(lockFile, '1');
   });
 
-  it.skip('can encrypt storage, and decrypt storage, but this time the fake one', async () => {
+  it('can encrypt storage, and decrypt storage, but this time the fake one', async () => {
```
Comment (Overtorment) [inline tests/e2e/bluewallet.spec.js:285]: muted test swapped.
why? what did you find?

**Review by Overtorment — CHANGES_REQUESTED** [review]
lgtm but theres a question about swapped test

File: tests/e2e/bluewallet.spec.js (hunk 1 above)
Comment (limpbrains) [reply tests/e2e/bluewallet.spec.js]: It's just a stupid error, fixed

**Review by Overtorment — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**